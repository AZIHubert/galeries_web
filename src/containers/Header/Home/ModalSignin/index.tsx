import { useFormik } from 'formik';
import * as React from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import Button from '#components/Button';
import Field from '#components/Field';
import Modal from '#components/Modal';
import RequiredField from '#components/RequiredField';
import TextSepatator from '#components/TextSeparator';

import { signinSchema } from '#helpers/schemas';

import {
  fetchSignin,
  resetSignin,
  setSignin,
} from '#store/actions';
import {
  loadingSelector,
  signinErrorSelector,
  signinStatusSelector,
} from '#store/selectors';

interface ModalSigninI {
  setCurrentEmail: React.Dispatch<React.SetStateAction<string>>;
  setCurrentModal: React.Dispatch<React.SetStateAction<HeaderModals | null>>;
}

const initialValues: form.SigninI = {
  confirmPassword: '',
  email: '',
  password: '',
  userName: '',
};

const ModalSignin = ({
  setCurrentEmail,
  setCurrentModal,
}: ModalSigninI) => {
  const dispatch = useDispatch();
  const formik = useFormik({
    initialValues,
    onSubmit: (values) => {
      if (!loading) {
        resetForm();
        dispatch(fetchSignin(values));
      }
    },
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: signinSchema,
  });
  const loading = useSelector(loadingSelector);
  const signinError = useSelector(signinErrorSelector);
  const signinStatus = useSelector(signinStatusSelector);

  React.useEffect(() => {
    if (signinStatus === 'success') {
      setCurrentEmail(formik.values.email);
      setCurrentModal('confirmLanding');
    }
  }, [signinStatus]);
  React.useEffect(() => () => resetForm(), []);

  const resetForm = () => {
    dispatch(resetSignin());
  };

  return (
    <Modal.Container
      containerTestId='signin'
    >
      <Button.Facebook
        action='signin'
      />
      <Button.Google
        action='signin'
      />
      <TextSepatator
        styles={{
          marginBottom: 9,
          marginTop: 9,
        }}
        stylesLaptopL={{
          marginBottom: 16,
          marginTop: 16,
        }}
        text='or'
      />
      <form
        onSubmit={formik.handleSubmit}
        data-testid='form'
      >
        <Field
          disabled={loading}
          error={
            formik.errors.userName || signinError.userName
          }
          fieldTestId='userName'
          id='userName'
          label='user name'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.userName) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  userName: '',
                },
              }));
            }
          }}
          required
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.userName}
          value={formik.values.userName}
        />
        <Field
          disabled={loading}
          error={
            formik.errors.email || signinError.email
          }
          fieldTestId='email'
          id='email'
          label='email'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.email) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  email: '',
                },
              }));
            }
          }}
          required
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.email}
          value={formik.values.email}
        />
        <Field
          disabled={loading}
          error={
            formik.errors.password || signinError.password
          }
          fieldTestId='password'
          id='password'
          label='password'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.password) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  password: '',
                },
              }));
            }
          }}
          required
          styles={{
            marginBottom: 6,
          }}
          stylesLaptopL={{
            marginBottom: 10,
          }}
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <Field
          disabled={loading}
          error={
            formik.errors.confirmPassword || signinError.confirmPassword
          }
          fieldTestId='confirmPassword'
          id='confirmPassword'
          label='confirm password'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.confirmPassword) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  confirmPassword: '',
                },
              }));
            }
          }}
          required
          styles={{
            marginBottom: 12,
          }}
          stylesLaptopL={{
            marginBottom: 15,
          }}
          touched={formik.touched.confirmPassword}
          type='password'
          value={formik.values.confirmPassword}
        />
        <RequiredField />
        <Button.Gradiant
          disabled={loading}
          styles={{
            marginBottom: 15,
            marginTop: 15,
          }}
          stylesLaptopL={{
            marginBottom: 22,
            marginTop: 22,
          }}
          type='submit'
          title='Sign in'
        />
      </form>
      <Button.Text
        disabled={loading}
        onClick={() => setCurrentModal('login')}
        styles={{
          fontSize: 0.65,
          justifyContent: 'center',
        }}
        stylesLaptopL={{
          fontSize: 0.8,
        }}
        text='You already have an account? click'
        textButton='here'
      />
    </Modal.Container>
  );
};

export default ModalSignin;
