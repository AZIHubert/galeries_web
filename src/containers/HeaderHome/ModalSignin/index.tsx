import { useFormik } from 'formik';
import * as React from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import FacebookButton from '#components/FacebookButton';
import Field from '#components/Field';
import GoogleButton from '#components/GoogleButton';
import GradientButton from '#components/GradientButton';
import ModalContainer from '#components/ModalContainer';
import RequiredField from '#components/RequiredField';
import TextButton from '#components/TextButton';
import TextSepatator from '#components/TextSeparator';

import { signinSchema } from '#helpers/schemas';

import {
  setSignin,
  fetchSignin,
} from '#store/actions';
import {
  loadingSelector,
  signinErrorSelector,
  signinStatusSelector,
} from '#store/selectors';

interface ModalSigninI {
  setCurrentEmail: React.Dispatch<React.SetStateAction<string>>;
  setCurrentModal: React.Dispatch<React.SetStateAction<HeaderModals | null>>;
}

const initialValues: form.SigninI = {
  confirmPassword: '',
  email: '',
  password: '',
  userName: '',
};

const ModalSignin = ({
  setCurrentEmail,
  setCurrentModal,
}: ModalSigninI) => {
  const dispatch = useDispatch();
  const formik = useFormik({
    initialValues,
    onSubmit: (values) => {
      if (!loading) {
        resetForm();
        dispatch(fetchSignin(values));
      }
    },
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: signinSchema,
  });
  const loading = useSelector(loadingSelector);
  const signinError = useSelector(signinErrorSelector);
  const signinStatus = useSelector(signinStatusSelector);

  React.useEffect(() => {
    if (signinStatus === 'success') {
      setCurrentEmail(formik.values.email);
      setCurrentModal('confirmLanding');
    }
  }, [signinStatus]);
  React.useEffect(() => () => resetForm(), []);

  const resetForm = () => {
    dispatch(setSignin({
      status: 'pending',
      errors: initialValues,
    }));
  };

  return (
    <ModalContainer>
      <FacebookButton
        action='signin'
      />
      <GoogleButton
        action='signin'
      />
      <TextSepatator
        marginBottom={9}
        marginBottomL={16}
        marginTop={9}
        marginTopL={16}
        text='or'
      />
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          error={
            formik.errors.userName || signinError.userName
          }
          id='userName'
          marginBottom={6}
          marginBottomL={10}
          label='user name'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.userName) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  userName: '',
                },
              }));
            }
          }}
          required
          touched={formik.touched.userName}
          value={formik.values.userName}
        />
        <Field
          disabled={loading}
          error={
            formik.errors.email || signinError.email
          }
          id='email'
          marginBottom={6}
          marginBottomL={10}
          label='email'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.email) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  email: '',
                },
              }));
            }
          }}
          required
          touched={formik.touched.email}
          value={formik.values.email}
        />
        <Field
          disabled={loading}
          error={
            formik.errors.password || signinError.password
          }
          id='password'
          marginBottom={6}
          marginBottomL={10}
          label='password'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.password) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  password: '',
                },
              }));
            }
          }}
          required
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <Field
          disabled={loading}
          error={
            formik.errors.confirmPassword || signinError.confirmPassword
          }
          id='confirmPassword'
          marginBottom={12}
          marginBottomL={15}
          label='confirm password'
          onBlur={formik.handleBlur}
          onChange={(e) => {
            formik.handleChange(e);
            if (signinError.confirmPassword) {
              dispatch(setSignin({
                errors: {
                  ...signinError,
                  confirmPassword: '',
                },
              }));
            }
          }}
          required
          touched={formik.touched.confirmPassword}
          type='password'
          value={formik.values.confirmPassword}
        />
        <RequiredField />
        <GradientButton
          disabled={loading}
          marginBottom={15}
          marginBottomL={22}
          marginTop={15}
          marginTopL={22}
          type='submit'
          title='Sign in'
        />
      </form>
      <TextButton
        disabled={loading}
        fontSize={0.65}
        fontSizeL={0.8}
        justifyContent='center'
        onClick={() => setCurrentModal('login')}
        text='You already have an account? click'
        textButton='here'
      />
    </ModalContainer>
  );
};

export default ModalSignin;
