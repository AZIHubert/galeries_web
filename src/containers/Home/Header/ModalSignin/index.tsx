import { useFormik } from 'formik';
import * as React from 'react';
import FacebookLogin, { ReactFacebookLoginInfo, ReactFacebookFailureResponse } from 'react-facebook-login';
import GoogleLogin from 'react-google-login';

import Field from '#components/Field';
import GradientButton from '#components/GradientButton';
import ModalContainer from '#components/ModalContainer';
import RequiredField from '#components/RequiredField';
import SocialMediaButton from '#components/SocialMediaButton';
import TextButton from '#components/TextButton';
import TextSepatator from '#components/TextSeparator';

import {
  loginFacebook,
  loginGoogle,
  signin,
} from '#helpers/api';

import { signinSchema } from '#helpers/schemas';

interface ModalSigninI {
  loading: boolean;
  setAccountCreate: React.Dispatch<React.SetStateAction<boolean>>;
  setError: React.Dispatch<React.SetStateAction<string>>;
  setCurrentEmail: React.Dispatch<React.SetStateAction<string>>;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  switchModal: () => void;
}

const initialValues = {
  confirmPassword: '',
  email: '',
  password: '',
  userName: '',
};

const ModalSignin = ({
  loading,
  setAccountCreate,
  setCurrentEmail,
  setError,
  setLoading,
  switchModal,
}: ModalSigninI) => {
  const responseFacebook = async (
    faceBookResponse: ReactFacebookLoginInfo | ReactFacebookFailureResponse,
  ) => {
    setLoading(true);
    try {
      const response = await loginFacebook(faceBookResponse);
      localStorage.setItem('authToken', response.data.token);
      localStorage.setItem('expiresIn', response.data.expiresIn);
      setLoading(false);
    } catch (err) {
      if (err.status === 500) {
        setError('something went wrong');
      } else {
        setError(err.data.errors);
      }
    }
    setLoading(false);
  };
  const responseGoogle = async (
    googleResponse: any,
  ) => {
    setLoading(true);
    try {
      const response = await loginGoogle(googleResponse.profileObj);
      localStorage.setItem('authToken', response.data.token);
      localStorage.setItem('expiresIn', response.data.expiresIn);
    } catch (err) {
      if (err.response.status === 500) {
        setError('something went wrong');
      } else {
        setError(err.response.data.errors);
      }
    }
    setLoading(false);
  };

  const formik = useFormik({
    initialValues,
    onSubmit: async (values, { setFieldError }) => {
      if (!loading) {
        setLoading(true);
        try {
          const response = await signin(values);
          console.log(response.data);
          setAccountCreate(true);
          setCurrentEmail(values.email);
          setLoading(false);
        } catch (err) {
          const { errors } = err.response.data;
          if (typeof errors === 'object') {
            Object.keys(errors).map((error) => setFieldError(error, errors[error]));
          }
          setLoading(false);
        }
      }
    },
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: signinSchema,
  });

  return (
    <ModalContainer
      testId='modalSignin'
    >
      <FacebookLogin
        appId="688539228486770"
        fields="email, gender, name, picture.type(large)"
        onClick={() => setLoading(true)}
        callback={responseFacebook}
        render={(renderProps) => (
          <SocialMediaButton
            action='signin'
            disabled={loading}
            marginBottom={12}
            onClick={renderProps.onClick}
          />
        )}
      />
      <GoogleLogin
        clientId="863840240633-tve0cuo6hib6uhgap61j1nkq03k7k5vq.apps.googleusercontent.com"
        buttonText="Login"
        onSuccess={responseGoogle}
        onFailure={(err) => console.log(err)}
        cookiePolicy={'single_host_origin'}
        render={(renderProps) => (
          <SocialMediaButton
            action='signin'
            disabled={loading}
            onClick={renderProps.onClick}
            variant='google'
          />
        )}
      />

      <TextSepatator
        marginBottom={12}
        marginTop={12}
        text='or'
      />
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          id='userName'
          error={formik.errors.userName}
          errorTestId='userNameError'
          fieldTestId='userNameField'
          marginBottom={7}
          label='user name'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.userName}
          value={formik.values.userName}
        />
        <Field
          disabled={loading}
          id='email'
          error={formik.errors.email}
          errorTestId='emailError'
          fieldTestId='emailField'
          marginBottom={7}
          label='email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.email}
          value={formik.values.email}
        />
        <Field
          disabled={loading}
          id='password'
          error={formik.errors.password}
          errorTestId='passwordError'
          fieldTestId='passwordField'
          marginBottom={7}
          label='password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <Field
          disabled={loading}
          id='confirmPassword'
          error={formik.errors.confirmPassword}
          errorTestId='confirmPasswordError'
          fieldTestId='confirmPasswordField'
          marginBottom={15}
          label='confirm password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.confirmPassword}
          type='password'
          value={formik.values.confirmPassword}
        />
        <RequiredField />
        <GradientButton
          testId='submitButton'
          disabled={loading}
          marginBottom={20}
          marginTop={20}
          type='submit'
          title='Sign in'
        />
      </form>
      <TextButton
        disabled={loading}
        fontSize={0.65}
        justifyContent='center'
        onClick={switchModal}
        testId='switchToLogin'
        text='You already have an account? click'
        textButton='here'
      />
    </ModalContainer>
  );
};

export default ModalSignin;
