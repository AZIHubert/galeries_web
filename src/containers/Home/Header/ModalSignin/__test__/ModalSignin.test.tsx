import * as React from 'react';
import renderer from 'react-test-renderer';
import {
  act,
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import ThemeProvider from '#contexts/ThemeContext';

import {
  CONFIRM_PASSWORD,
  EMAIL_FIELD,
  HAS_SPACES,
  MAX_LENGTH_THRITY,
  MIN_LENGTH_OF_HEIGH,
  MIN_LENGTH_OF_THREE,
  PASSWORD,
  REQUIRED,
} from '#helpers/formErrors';

import ModalSignin from '../index';

const Container = () => {
  const [loading, setLoading] = React.useState<boolean>(false);
  const [accountCreate, setAccountCreate] = React.useState<boolean>(false);
  const switchModal = () => {};
  const setCurrentEmail = () => {};
  return (
    <ThemeProvider>
      <ModalSignin
        loading={loading}
        setLoading={setLoading}
        setAccountCreate={setAccountCreate}
        switchModal={switchModal}
        setCurrentEmail={setCurrentEmail}
      />
      {accountCreate
        ? (
          <p
            data-testid='accountCreate'
          >
            accountCreate
          </p>
        )
        : null}
    </ThemeProvider>
  );
};

describe('ModalSignin', () => {
  const mockedSetLoading = jest.fn;
  const mockedSetAccountCreate = jest.fn;
  const mockedSwitchModal = jest.fn;
  const mockedSetCurrentEmail = jest.fn;
  let confirmPasswordField: HTMLElement;
  let emailField: HTMLElement;
  let submitButton: HTMLElement;
  let passwordField: HTMLElement;
  let userNameField: HTMLElement;
  beforeEach(() => {
    act(() => {
      render(<Container />);
    });
    confirmPasswordField = screen.getByTestId('confirmPasswordField');
    emailField = screen.getByTestId('emailField');
    passwordField = screen.getByTestId('passwordField');
    submitButton = screen.getByTestId('submitButton');
    userNameField = screen.getByTestId('userNameField');
  });
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <ModalSignin
          loading={false}
          setLoading={mockedSetLoading}
          setAccountCreate={mockedSetAccountCreate}
          switchModal={mockedSwitchModal}
          setCurrentEmail={mockedSetCurrentEmail}
        />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not loading if errors', async () => {
    fireEvent.click(submitButton);
    const confirmPasswordError = await screen.findByTestId('confirmPasswordError');
    const emailError = await screen.findByTestId('emailError');
    const passwordError = await screen.findByTestId('passwordError');
    const userNameError = await screen.findByTestId('userNameError');
    expect(confirmPasswordError).toHaveTextContent(REQUIRED);
    expect(confirmPasswordField).not.toBeDisabled();
    expect(emailError).toHaveTextContent(REQUIRED);
    expect(emailField).not.toBeDisabled();
    expect(passwordError).toHaveTextContent(REQUIRED);
    expect(passwordField).not.toBeDisabled();
    expect(submitButton).toHaveTextContent('Sign in');
    expect(submitButton).not.toBeDisabled();
    expect(userNameError).toHaveTextContent(REQUIRED);
    expect(userNameField).not.toBeDisabled();
  });
  it('should loading if no error', async () => {
    fireEvent.change(confirmPasswordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(emailField, { target: { value: 'user@email.com' } });
    fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(userNameField, { target: { value: 'user' } });
    fireEvent.click(submitButton);
    await screen.findAllByText('loading');
    expect(confirmPasswordField).toBeDisabled();
    expect(emailField).toBeDisabled();
    expect(passwordField).toBeDisabled();
    expect(submitButton).toBeDisabled();
    expect(submitButton).toBeDisabled();
    expect(userNameField).toBeDisabled();
  });
  it('should set accountCreated to true if no error', async () => {
    fireEvent.change(confirmPasswordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(emailField, { target: { value: 'user@email.com' } });
    fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(userNameField, { target: { value: 'user' } });
    fireEvent.click(submitButton);
    const accountCreate = await screen.findByTestId('accountCreate');
    expect(accountCreate).toHaveTextContent('accountCreate');
  });
  describe('should show an error if', () => {
    describe('userName field', () => {
      it('is empty', async () => {
        fireEvent.blur(userNameField);
        const userNameError = await screen.findByTestId('userNameError');
        expect(userNameError).not.toBeNull();
        expect(userNameError).toHaveTextContent(REQUIRED);
      });
      it('contain spaces', async () => {
        fireEvent.change(userNameField, { target: { value: 'a a' } });
        fireEvent.blur(userNameField);
        const userNameError = await screen.findByTestId('userNameError');
        expect(userNameError).not.toBeNull();
        expect(userNameError).toHaveTextContent(HAS_SPACES);
      });
      it('is less than three char', async () => {
        fireEvent.change(userNameField, { target: { value: 'aa' } });
        fireEvent.blur(userNameField);
        const userNameError = await screen.findByTestId('userNameError');
        expect(userNameError).not.toBeNull();
        expect(userNameError).toHaveTextContent(MIN_LENGTH_OF_THREE);
      });
      it('is more than thirty char', async () => {
        fireEvent.change(userNameField, { target: { value: 'a'.repeat(31) } });
        fireEvent.blur(userNameField);
        const userNameError = await screen.findByTestId('userNameError');
        expect(userNameError).not.toBeNull();
        expect(userNameError).toHaveTextContent(MAX_LENGTH_THRITY);
      });
    });
    describe('email field', () => {
      it('is empy', async () => {
        fireEvent.blur(emailField);
        const emailError = await screen.findByTestId('emailError');
        expect(emailError).not.toBeNull();
        expect(emailError).toHaveTextContent(REQUIRED);
      });
      it('is not an email', async () => {
        fireEvent.change(emailField, { target: { value: 'notAnEmail' } });
        fireEvent.blur(emailField);
        const emailError = await screen.findByTestId('emailError');
        expect(emailError).not.toBeNull();
        expect(emailError).toHaveTextContent(EMAIL_FIELD);
      });
    });
    describe('password field', () => {
      it('is empty', async () => {
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(REQUIRED);
      });
      it('contain spaces', async () => {
        fireEvent.change(passwordField, { target: { value: 'a a' } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(HAS_SPACES);
      });
      it('contain less than 8 chars', async () => {
        fireEvent.change(passwordField, { target: { value: 'a'.repeat(7) } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(MIN_LENGTH_OF_HEIGH);
      });
      it('contain more than 30 chars', async () => {
        fireEvent.change(passwordField, { target: { value: 'a'.repeat(31) } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(MAX_LENGTH_THRITY);
      });
      it('doesn\'t contain any uppercase', async () => {
        fireEvent.change(passwordField, { target: { value: 'aaoudjiuvhds9!' } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(PASSWORD);
      });
      it('doesn\'t contain any lowercase', async () => {
        fireEvent.change(passwordField, { target: { value: 'AAOUDJIUVHDS9!' } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(PASSWORD);
      });
      it('doesn\'t contain any number', async () => {
        fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds!' } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(PASSWORD);
      });
      it('doesn\'t contain any special char', async () => {
        fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds9' } });
        fireEvent.blur(passwordField);
        const passwordError = await screen.findByTestId('passwordError');
        expect(passwordError).not.toBeNull();
        expect(passwordError).toHaveTextContent(PASSWORD);
      });
    });
    describe('confirm password', () => {
      it('and password not match', async () => {
        fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds9!' } });
        fireEvent.change(confirmPasswordField, { target: { value: 'wrongPassword' } });
        fireEvent.blur(confirmPasswordField);
        const confirmPasswordError = await screen.findByTestId('confirmPasswordError');
        expect(confirmPasswordError).not.toBeNull();
        expect(confirmPasswordError).toHaveTextContent(CONFIRM_PASSWORD);
      });
    });
  });
});
