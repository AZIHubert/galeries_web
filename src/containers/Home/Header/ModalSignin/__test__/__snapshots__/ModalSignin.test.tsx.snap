// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSignin renders without crashing 1`] = `
<div>
  <button
    data-testid="button"
    disabled={false}
    onClick={[Function]}
  >
    <img
      alt="logo facebook"
      src=""
    />
    sign in 
    with Facebook
  </button>
  <button
    data-testid="button"
    disabled={false}
    onClick={[Function]}
  >
    <img
      alt="logo google"
      src=""
    />
    sign in 
    with Google
  </button>
  <div>
    or
  </div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="userName"
    >
      user name *
    </label>
    <input
      data-testid="userNameField"
      disabled={false}
      id="userName"
      name="userName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="email"
    >
      email *
    </label>
    <input
      data-testid="emailField"
      disabled={false}
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="password"
    >
      password *
    </label>
    <input
      data-testid="passwordField"
      disabled={false}
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
      value=""
    />
    <label
      htmlFor="confirmPassword"
    >
      confirm password *
    </label>
    <input
      data-testid="confirmPasswordField"
      disabled={false}
      id="confirmPassword"
      name="confirmPassword"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
      value=""
    />
    <div>
      * Required field
    </div>
    <button
      data-testid="submitButton"
      disabled={false}
      type="submit"
    >
      signin
    </button>
  </form>
</div>
`;
